### PR TITLE
Add sync and async doc in README for helloworld

### PR DESCRIPTION
**Pull request by yang-g**

### CODE REVIEW

File: cpp/helloworld/README.md [hunk 1]
```diff
@@ -89,8 +89,172 @@ $ protoc -I ../../protos/ --cpp_out=. ../../protos/helloworld.proto
 
 ### Writing a client
 
-This is an incomplete tutorial. For now the reader should refer to [greeter_client.cc](https://github.com/grpc/grpc-common/blob/master/cpp/helloworld/greeter_client.cc).
+- Create a channel. A channel is a logical connection to an endpoint. A gRPC
+  channel can be created with the target address, credentials to use and
+  arguments as follows
+
+    ```
+    auto channel = CreateChannel("localhost:50051", InsecureCredentials(), ChannelArguments());
```
Comment (a11r) [inline cpp/helloworld/README.md]: Please drop ChannelArguments from the example once the corresponding code cahnge is done. OK to do this in a separate PR.

File: cpp/helloworld/README.md [hunk 2]
```diff
@@ -89,8 +89,172 @@ $ protoc -I ../../protos/ --cpp_out=. ../../protos/helloworld.proto
...
+
+For a working example, refer to [greeter_server.cc](https://github.com/grpc/grpc-common/blob/master/cpp/helloworld/greeter_server.cc).
+
+### Writing asynchronous client and server
+
+gRPC uses `CompletionQueue` API for asynchronous operations. The basic work flow
+is
+- bind a `CompletionQueue` to a rpc call
+- do something like a read or write, present with a unique `void*` tag
+- call `CompletionQueue::Next` to poll the events. If the tag appears, the
```
Comment (a11r) [inline cpp/helloworld/README.md]: Suggested rephrasing:
- call `CompletionQueue::Next` to wait for operations to complete. When a tag appears, it indicates that the corresponding operation is complete.

File: cpp/helloworld/README.md [hunk 3]
```diff
@@ -89,8 +89,172 @@ $ protoc -I ../../protos/ --cpp_out=. ../../protos/helloworld.proto
...
+  previous operation finishes.
+
+#### Async client
+
+The channel and stub creation code is the same as the sync client.
+
+- Initiate the rpc and create a handle for the rpc. Bind a `CompletionQueue` to
```
Comment (a11r) [inline cpp/helloworld/README.md]: Bind the rpc to a `CompletionQueue`.

File: cpp/helloworld/README.md [hunk 4]
```diff
@@ -89,8 +89,172 @@ $ protoc -I ../../protos/ --cpp_out=. ../../protos/helloworld.proto
...
+    ```
+
+- Ask for reply and final status, with a unique tag
+
+    ```
+    Status status;
+    rpc->Finish(&reply, &status, (void*)1);
+    ```
+
+- Poll the completion queue for the tag. The reply and status are ready once the
```
Comment (a11r) [inline cpp/helloworld/README.md]: - Wait for the completion queue to return the next tag.  The reply and status are ready once the tag passed into the corresponding `Finish()`call is returned.

**Comment by a11r** [comment]
Added some suggested edits.

File: cpp/helloworld/README.md (hunk 1 above)
Comment (yang-g) [inline cpp/helloworld/README.md]: ack.

File: cpp/helloworld/README.md (hunk 2 above)
Comment (yang-g) [inline cpp/helloworld/README.md]: Done.

File: cpp/helloworld/README.md (hunk 3 above)
Comment (yang-g) [inline cpp/helloworld/README.md]: Done.

File: cpp/helloworld/README.md (hunk 4 above)
Comment (yang-g) [inline cpp/helloworld/README.md]: Done. All others changed accordingly.

**Comment by yang-g** [comment]
PTAL. Thanks
